module/zfs/zvol.c

```diff
@@ -2239,18 +2236,31 @@ zvol_set_snapdev_sync_cb(dsl_pool_t *dp, dsl_dataset_t *ds, void *arg)
 }
 
 /*
- * Traverse all child snapshot datasets and apply snapdev appropriately.
+ * Traverse all child datasets and apply snapdev appropriately.
+ * We call dsl_prop_set_sync_impl() here to set the value only on the toplevel
```

Comment:
What is the use case you are referring to (please give the command sequence) ?
Please provide more detail.